src/dict.c

```diff
@@ -676,22 +693,36 @@ void dictRelease(dict *d)
 dictEntry *dictFind(dict *d, const void *key)
 {
     dictEntry *he;
-    uint64_t h, idx, table;
+    uint64_t h, idx;
 
     if (dictSize(d) == 0) return NULL; /* dict is empty */
-    if (dictIsRehashing(d)) _dictRehashStep(d);
+
     h = dictHashKey(d, key);
-    for (table = 0; table <= 1; table++) {
-        idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[table]);
-        if (table == 0 && (long)idx < d->rehashidx) continue; 
-        he = d->ht_table[table][idx];
-        while(he) {
-            void *he_key = dictGetKey(he);
-            if (key == he_key || dictCompareKeys(d, key, he_key))
-                return he;
-            he = dictGetNext(he);
+    idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[0]);
+    he = d->ht_table[0][idx];
```

Comment:
I wasn't closely following the internal thread we had on this PR, but `d->ht_table[0][idx];` is a random DRAM access that may be avoidable if `idx >= d->rehashidx` is true, since the item **must** be in the second table since it **must** have been rehashed. I remember posting a suggestion where we first check if the item can only be in the second table. If it is, we do one lookup. Otherwise we check the first and fallback to the second.